ast/sql.go

```diff
@@ -941,64 +950,108 @@ func (d *DropRole) SQL() string {
 
 func (g *Grant) SQL() string {
 	sql := "GRANT "
-	if g.Privileges != nil {
-		sql += string(g.Privileges[0].Name)
-		if g.Privileges[0].Columns != nil {
-			sql += "(" + CommaSeparatedSQL(g.Privileges[0].Columns) + ")"
-		}
-		for _, priv := range g.Privileges[1:] {
-			sql += ", " + string(priv.Name)
-			if priv.Columns != nil {
-				sql += "(" + CommaSeparatedSQL(priv.Columns) + ")"
-			}
-		}
-		sql += " ON TABLE " + CommaSeparatedSQL(g.TableNames)
-	} else {
-		sql += "ROLE " + CommaSeparatedSQL(g.GrantRoleNames)
+	sql += g.Privilege.SQL()
+	sql += " TO ROLE " + g.Roles[0].SQL()
```

Comment:
Not sure if we should use `Strings.Join` to simplify those concat statements.

Reply:
Spanner SQL is not ours but is Google's. That is, we cannot know the complete picture of it. Without the complete view, we cannot find common parts of the code actually. If we find code that looks like common parts but Google disagrees with it, the commonalities we introduced are wasted. Therefore, we did not use common utilities in `ast` and `parser`.

Reply:
Sorry for the wrong words. What I mean is that maybe we can use `Strings.Join` to concat the SQL instead of for-loop.

Reply:
Thank you!